include/nuttx/lib/math32.h

```diff
@@ -52,11 +52,15 @@
 #define FLS2(n)  ((n) & 0x2        ?  1 + FLS1 ((n) >>  1) : FLS1 (n))
 #define FLS1(n)  ((n) & 0x1        ?  1 : 0)
 
+/* Checks if an integer is power of two at compile time */
+
+#define IS_POWER_OF_2(n)           ((n) > 0 && ((n) & (n - 1)) == 0)
+
 /* Returns round up and round down value of log2(n). Note: it can be used at
  * compile time.
  */
 
-#define LOG2_CEIL(n)  ((n) & (n - 1) ? FLS(n) : FLS(n) - 1)
+#define LOG2_CEIL(n)  (IS_POWER_OF_2(n) ? FLS(n) - 1 : FLS(n))
```

Comment:
I think this change can be dropped now

Reply:
@pkarashchenko can we keep it so that it may help other cases? or do you have other concerns with it?

Reply:
In general I recommend to add only necessary changes if possible. Especially that updated `LOG2_CEIL` macro leads to a different behavior than before (`n > 0` is not in the current code). Also `IS_POWER_OF_2` macro seems to produce a `false` result if `n` is `1` (`2^0=1`), I'm not sure if the intention was to cover this case, but if somebody will use a new macro without looking into implementation it can lead to unexpected results.

Reply:
I do not mind keeping the new macro if it brings the value. Just please test the behavior of `LOG2_CEIL(0);`.

Reply:
log(0) isn't defined behavior, so I think it's fine to change the result in this case.

Reply:
log2(0), but this is the closest to the parameter upper log2 boundary. Like log2_ceil(3) is 2, log2_ceil(5) is 3, so the closest log2_ceil(0) seems to be 0 (2^0), isn't it?

Reply:
@pkarashchenko my observations:
- Tests for `LOG2CEIL(0)`: previously it returns `-1` and now it returns `0`.
- Python simply reports domain error for `ceil(log2(0))`.
- There are 3 uses of it in upstream: two with `size_t` input, one with OPEN_MAX.

Reply:
Python error is obvious as log2(0) is undefined. I'm fine to keep the implementation from this PR.